From cc650803d1402fe1b166f87be7bec582b97f27b6 Mon Sep 17 00:00:00 2001
From: Hariprasad Kelam <hkelam@marvell.com>
Date: Wed, 12 Feb 2020 16:05:40 +0530
Subject: [PATCH 479/767] octeontx2-af: Update tx parse nibble config

commit 8d28e5a13e0d25686532da84a8efd2efb744d3c2 from
git@git.assembla.com:cavium/WindRiver.linux.git

Few silicons need rx/tx parse nibble config to be same.
Update tx parse nibble config based on silicon version.

Change-Id: Idafdc09d04853f9f070255e182afb19689e1cfe5
Signed-off-by: Hariprasad Kelam <hkelam@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/23277
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/rvu_npc.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/rvu_npc.c b/drivers/net/ethernet/marvell/octeontx2/af/rvu_npc.c
index c209e3f0636b..a6e342e494b0 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/rvu_npc.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/rvu_npc.c
@@ -1476,6 +1476,7 @@ int rvu_npc_init(struct rvu *rvu)
 	if (nibble_ena) {
 		tx_kex &= ~NPC_PARSE_NIBBLE;
 		tx_kex |= FIELD_PREP(NPC_PARSE_NIBBLE, nibble_ena);
+		npc_mkex_default.keyx_cfg[NIX_INTF_TX] = tx_kex;
 	}
 	rvu_write64(rvu, blkaddr, NPC_AF_INTFX_KEX_CFG(NIX_INTF_TX), tx_kex);
 
-- 
2.31.1

